src/utils/response/rawApiResponse.ts

```diff
@@ -0,0 +1,33 @@
+import { AxiosResponse } from 'axios';
+
+export type RawApiResponse = {
+  Host?: string;
+  Codeversion?: string;
+  Success?: boolean;
+  SuccessNested?: boolean;
+  NumResults?: number;
+  NumResultsNested?: number;
+  ResourceName?: string;
+  RequestStartTime?: string;
+  RequestEndTime?: string;
+  ServerTimeZone?: string;
+  Results?: any;
+  Deleted?: any;
+  Affected?: any;
+};
+
+export async function resolveAxiosResponse(promise: Promise<AxiosResponse<any>>): Promise<RawApiResponse> {
+  const resolved = await promise;
+
+  if (resolved === undefined) {
```

Comment:
I believe the promise above should be properly typed as `Promise<AxiosResponse<RawApiResponse>>`. I don't believe it should ever resolve to undefined (that would be something we should make sure never happens, in case of an error it needs to reject), therefore this if should be removed

Reply:
promise is never of type Promise<AxiosResponse<RawApiResponse>>